Assets/Scripts/LogicServices/Algorithm/Randomness.cs

```diff
@@ -0,0 +1,19 @@
+using UnityEngine;
+
+namespace LogicServices.Algorithm
+{
+    public class Randomness
+    {
+        private readonly float _changeChance;
+
+        public Randomness(float changeChance)
```

Comment:
Why is the chance set in the constructor? When it needs to be able to change based on the specific decision that is being made. E.i. self building chance = 20%, effect is 30%